src/utilities/new_versions.jl

```diff
@@ -99,3 +99,221 @@ function skip_equality_specifiers(
            !contains(compat_entry_verbatim, '>') &&
            !contains(compat_entry_verbatim, '<')
 end
+
+function create_new_pull_request(
+    api::GitLab.GitLabAPI,
+    repo::GitLab.Project,
+    new_branch_name::AbstractString,
+    master_branch_name::AbstractString,
+    title::AbstractString,
+    body::AbstractString,
+)
+    return create_pull_request(
+        api,
+        repo.owner.name,
+        repo.name;
+        id=repo.id,
+        source_branch=new_branch_name,
+        target_branch=master_branch_name,
+        title=title,
+        body=body,
+    )
+end
+
+function create_new_pull_request(
+    api::GitHub.GitHubAPI,
+    repo::GitHub.Repo,
+    new_branch_name::AbstractString,
+    master_branch_name::AbstractString,
+    title::AbstractString,
+    body::AbstractString,
+)
+    return create_pull_request(
+        api,
+        repo.owner.login,
+        repo.name;
+        id=repo.id,
+        source_branch=new_branch_name,
+        target_branch=master_branch_name,
+        title=title,
+        body=body,
+    )
+end
+
+function get_url_with_auth(
+    api::GitHub.GitHubAPI, hostname::AbstractString, repo::GitHub.Repo
+)
+    return "https://x-access-token:$(api.token.token)@$hostname/$(repo.full_name).git"
+end
+
+function get_url_for_ssh(::GitHub.GitHubAPI, hostname::AbstractString, repo::GitHub.Repo)
+    return "git@$hostname:$(repo.full_name).git"
+end
+
+function get_url_with_auth(
+    api::GitLab.GitLabAPI, hostname::AbstractString, repo::GitLab.Project
+)
+    return "https://oauth2:$(api.token.token)@$hostname/$(repo.path_with_namespace).git"
+end
+
+function get_url_for_ssh(::GitLab.GitLabAPI, hostname::AbstractString, repo::GitLab.Project)
+    return "git@$(hostname):$(repo.path_with_namespace).git"
+end
+
+function continue_with_pr(
+    dep::DepInfo,
+    bump_compat_containing_equality_specifier::Bool,
+)
+   # Determine if we need to make a new PR
+    if (!isnothing(dep.latest_version) && !isnothing(dep.version_spec)) &&
+       dep.latest_version in dep.version_spec
+        @info(
+            "latest_version in version_spec",
+            dep.latest_version,
+            dep.version_spec,
+            dep.name,
+            dep,
+        )
+        return false
+    elseif skip_equality_specifiers(
+        bump_compat_containing_equality_specifier, dep.version_verbatim
+    )
+        @info(
+            "Skipping compat entry because it contains an equality specifier",
+            dep.version_verbatim,
+            dep.name,
+            dep,
+        )
+        return false
+    elseif isnothing(dep.latest_version)
+        @error("The dependency was not found in any of the registries", dep.name, dep)
+        return false
+    end
+    return true
+end
+
+
+function make_pr_for_new_version(
```

Comment:
Maybe a few comments in here explaining what each layered block of code is doing? For future maintenance